### PR TITLE
PM-10630 setup autofill UI and interactions set up

### PR DESCRIPTION
**Pull request by dseverns-livefront**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-10630

## 📔 Objective
- Setup Auto-Fill screen UI set and ready to be integrated with the new Onboarding flow
- VM functions setup
- Dialogs in place

## 📸 Screenshots
image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupAutofillScreen.kt [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+            color = MaterialTheme.colorScheme.onSurfaceVariant,
+            textAlign = TextAlign.Center,
+            modifier = Modifier
+                .standardHorizontalMargin()
+                .align(Alignment.CenterHorizontally),
+        )
+        Spacer(modifier = Modifier.height(24.dp))
+        BitwardenWideSwitch(
+            label =
+            stringResource(
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupAutofillScreen.kt:167]: Can we move this up a line

File: app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupAutoFillViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class SetupAutoFillViewModelTest : BaseViewModelTest() {
+
+    private val mutableAutoFillEnabledStateFlow = MutableStateFlow(false)
+    private val settingsRepository = mockk<SettingsRepository>(relaxed = true) {
+        every { isAutofillEnabledStateFlow } returns mutableAutoFillEnabledStateFlow
+        every { disableAutofill() } answers {
+            mutableAutoFillEnabledStateFlow.value = false
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupAutoFillViewModelTest.kt:25]: Doesn't this overcomplicate the test. Could this just be `every { disableAutofill() } just runs`

File: app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupAutofillScreenTest.kt [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class SetupAutofillScreenTest : BaseComposeTest() {
+
+    private var onNavigateToCompleteSetupCalled = false
+    private val mutableEventFlow = bufferedMutableSharedFlow<SetupAutoFillEvent>()
+    private val mutableStateFlow =
+        MutableStateFlow(SetupAutoFillState(dialogState = null, autofillEnabled = false))
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupAutofillScreenTest.kt:28]: Can we define the state down below as a constant `DEFAULT_STATE`

File: app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupAutoFillViewModelTest.kt (hunk 2 above)
Comment (dseverns-livefront) [reply app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupAutoFillViewModelTest.kt:25]: I thought this was pretty clever and allowed for a little more full circle test, but I can nuke it so the test is just the unit of "is the mock called"

**Review by david-livefront — DISMISSED** [review]

**Review by david-livefront — APPROVED** [review]

**Pull request merged**